clients/src/main/java/org/apache/kafka/common/protocol/Errors.java

```diff
@@ -495,7 +496,14 @@ public ApiException build(String message) {
             public ApiException build(String message) {
                 return new OperationNotAttemptedException(message);
             }
-        });
+    }),
+    KAFKA_STORAGE_ERROR(56, "Disk error when trying to access log file on the disk.",
+        new ApiExceptionBuilder() {
+            @Override
+            public ApiException build(String message) {
+                return new KafkaStorageException(message);
+            }
+    });
```

Comment:
Good point. I added the following comment:

```
 * Note that client library will convert an unknown error code to the non-retriable UnknownServerException if the client library
 * version is old and does not recognize the newly-added error code. Therefore when a new server-side error is added,
 * we may need extra logic to convert the new error code to another existing error code before sending the response back to
 * the client if the request version suggests that the client may not recognize the new error code.
```

I also added the following comment in `KafkaStorageException.java`:

```
 * Here are the guidelines on how to handle KafkaStorageException and IOException:
 *
 * 1) If the server has not finished loading logs, IOException should not be converted to KafkaStorageException
 * 2) After the server has finished loading logs, IOException should be caught and trigger LogDirFailureChannel.maybeAddLogFailureEvent
 *    Then the IOException should either be swallowed and logged, or be converted to KafkaStorageException.
 * 3) It is preferred for IOException to be caught in Log rather than in ReplicaManager or LogSegment.
```